jest/hasteImpl.js

```diff
@@ -37,7 +37,7 @@ const NAME_REDUCERS /*: Array<[RegExp, string]> */ = [
   // strip .js/.js.flow suffix
   [/^(.*)\.js(\.flow)?$/, '$1'],
   // strip .android/.ios/.native/.web suffix
-  [/^(.*)\.(android|ios|native|web|windows|dom)$/, '$1'],
+  [/^[^\.]+(\.es\d)?$/, '$1'],
```

Comment:
Seems like this could use a comment, not sure how it works

Reply:
I ran a test on the regex, and I can't get it to match `Foo.native.js`. This was code that I had commented in Slack about. Unfortunately, `hasteImpl` is dynamically required by `jest-haste-map`, so it's not trivial to inject "plugin" platform extensions.

However, we can move to a model where plugin platforms must create their own `metro.config.js` and `hasteImpl.js` file with the platform extensions in the regex for `getHasteName`, but that would mean it would be more challenging to add plugin platforms to projects that already have their own `metro.config.js`. We could probably use `babel` to modify existing `metro.config.js` files and add `hasteImplModulePath` if it's not already there, and just throw a warning if it's already there pointing to instructions on how to add the necessary platform behavior to their `hasteImpl.js`.

Reply:
It looks like this regex aims to strip out `.es6` suffixes, but it needs one more capture group. As it stands right now, it would take `index.es6` and return `.es6`, which is unlike the other reducers that strip out `.js` or `.flow.js`.

It looks like you want to use `^([^\.]+)(\.es\d)?$` here, which would return `index` in the above example.

This doesn't match the comment above, however.